### PR TITLE
nvidia 535.161.07

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

The current version of the Nvidia drivers is not recent enough for the A100s we get on AWS.

 https://github.com/determined-ai/lore/blob/4a131a94f5cd0721f4de1cf42190605dfcc37abd/deployment/test-genai-env.py#L1 
Our test fails when it's an a100 but passes for v100 (fine after the update)

test instances:
https://us-west-2.console.aws.amazon.com/ec2/home?region=us-west-2#InstanceDetails:instanceId=i-0c336407da0d6e476
https://us-west-2.console.aws.amazon.com/ec2/home?region=us-west-2#InstanceDetails:instanceId=i-0e30ea3573a00b7f9
which are running:
| NVIDIA-SMI 535.86.05              Driver Version: 535.86.05    CUDA Version: 12.2     | (edited) 

Which driver version works on A100?

Hamid Zare
  [18:18](https://hpe-aiatscale.slack.com/archives/D05DLR27NP9/p1709165913633089)
| NVIDIA-SMI 535.161.07             Driver Version: 535.161.07   CUDA Version: 12.2     |
seems to work fine

## Checklist

- [ ] Bump VERSION to make the pushed images are tagged with the right version.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] Test the images by running the test `bumpenvs` procedure in the determined repo. See [README](https://github.com/determined-ai/environments/blob/main/README.md).

### CODE REVIEW

**Comment by hamidzr** [comment]
nvidia version launched on an agent with this update:
`I-0CB4E374AC3B962A7`
```
nubuntu@ip-10-192-10-81:~$ nvidia-smi
Thu Feb 29 15:05:06 2024
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.161.07             Driver Version: 535.161.07   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A100-SXM4-40GB          On  | 00000000:10:1C.0 Off |                    0 |
| N/A   37C    P0              53W / 400W |      0MiB / 40960MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+
```

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.27.1
+0.29.1
```
Comment (hamidzr) [inline VERSION:1]: I'm not sure about how we update these

**Comment by hamidzr** [comment]
nvidia mentions it supports A100s in the older version here https://docs.nvidia.com/datacenter/tesla/tesla-release-notes-535-86-10/#fixed-issues although it's a .10 vs our .6 in the patch segment

**Comment by hamidzr** [comment]
running through the dry run  https://github.com/determined-ai/environments/blob/95c7a14e45138a403479a056968d97b578443268/README.md#L31

https://github.com/determined-ai/determined/tree/hz-new-ami

**Comment by ioga** [comment]
>it's looking like this version of the drivers that come with the latest AMIs are not good enough for A100s

I thought we've had people running A100s on AWS for a long time? and Lore itself has been running on A100s for a while? 

updating drivers is always good but if there's indeed an issue "A100 does not work" I'd be very surprised.

File: VERSION (hunk 1 above)
Comment (ioga) [reply VERSION:1]: current release in progress is 0.29.0, so this seems right.

**Review by tayritenour — COMMENTED** [review]
It's currently unclear to me whether this is necessary, we did see yesterday that newly spun up a100 machines were not behaving with our current AMIs. If we can confirm with a newly built AMI using this version that it's okay I approve but I'm not sure we fully did that yet

**Comment by hamidzr** [comment]
> I thought we've had people running A100s on AWS for a long time? and Lore itself has been running on A100s for a while?

those were mainly old agents that people have had manually updated drivers for, some of the clusters had startup scripts to do this + bunch of other things that I think lead to the new ami and a100 instance combination issue not being detected: we didn't have a good limited test to check if the drivers are good enough and the new deployments used different newer ami and the testing we did was mostly limited to other GPU types since each time we couldn't get new A100s and we didn't catch this.

**Comment by hamidzr** [comment]
> It's currently unclear to me whether this is necessary, we did see yesterday that newly spun up a100 machines were not behaving with our current AMIs. If we can confirm with a newly built AMI using this version that it's okay I approve but I'm not sure we fully did that yet

yesterday we tested the newest ami and this test only passed for other gpu types but not A100s
https://github.com/determined-ai/lore/blob/3093213045ab75b07e3579ba0ccb738e08da4c38/deployment/test-genai-env.py#L1 right? without manually updated drivers.
the same test passes on instances spun up with these amis and no manual updates on a100s (the cluster and instance ids are linked up) I tested model load as well.

**Review by tayritenour — APPROVED** [review]
LGTM.

We are able to use this new AMI to install without custom drivers being installed in the `startup_script` for the resource pool. Before we had to make sure to use a bunch of manual steps to make a100's work

**Comment by hamidzr** [comment]
two ray tests are failing with a dry run update. looking into those before proceeding https://hpe-aiatscale.slack.com/archives/C04C9JXB1C2/p1709229743548879

**Review by MikhailKardash — APPROVED** [review]
LGTM. Just run this by product.

**Comment by hamidzr** [comment]
the dry run on determined passed all the ci and longrunning jobs https://app.circleci.com/pipelines/github/determined-ai/determined/51854

**Pull request merged**